sys/ztimer/periph_rtt.c

```diff
@@ -50,7 +51,8 @@ static void _ztimer_periph_rtt_set(ztimer_clock_t *clock, uint32_t val)
     }
 
     unsigned state = irq_disable();
-    rtt_set_alarm(rtt_get_counter() + val, _ztimer_periph_rtt_callback, clock);
+    /* warning: RTT_MAX_VALUE needs to be (2^n) - 1 */
```

Comment:
This requirement should either be asserted in the init and/or be documented somewhere more obvious. Otherwise it is going to lead to hard to debug issues when a RTT peripheral doesn't adhere to this.

Reply:
I'm not sure if an assert here is effort well spend. It is very likely that the limit comes from the size of registers holding the current timer value and the next alarm value. And unless some strange stuff like BCD encoding is used, the maximum value will just naturally be `(2^n) - 1`.

And if someone indeed starts producing RTTs with BCD encoding internally used, I'd say it is fine to just never support those.

But it clearly wouldn't hurt to document this requirement. The lowest value could also be documented. But it should be equally unlikely that someone allows negative values in an RTT. (That would make some sense for the year register in an RTC, but I guess no one in her/his right mind would use signed counters here.)

Reply:
hm did github eat my reply? I agree that non-`(2^n) - 1` is unlilely. I've added a static_assert anyways.

Reply:
I was a bit puzzled why gcc doesn't do this on its own when specifying plain modulo.

Reply:
:+1: for static assert. (I misinterpreted the suggestion as adding a plain assert() with runtime cost)

Reply:
> +1 for static assert. (I misinterpreted the suggestion as adding a plain assert() with runtime cost)

Well, I would be fine with an assert in the init code, but not in the `_ztimer_periph_rtt_set` function